Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -118,10 +123,17 @@ public string RedactAndStore(string connectionString)
         /// value for the key, also checks if a connection to the IotHub
         /// can be made.
         /// </summary>
-        public void ValidateConnectionString(string connectionString)
+        public static async Task ValidateConnectionString(string connectionString)
         {
-            var match = Regex.Match(connectionString, CONNSTRING_REGEX);
+            // valid if default or null
+            if (connectionString == ServicesConfig.USE_DEFAULT_IOTHUB ||
+                connectionString.IsNullOrWhiteSpace())
+            {
+                return;
+            }
 
+            // check format of provided string
+            var match = Regex.Match(connectionString, CONNSTRING_REGEX);
             if (!match.Success)
```

Comment:
consider splitting the method, e.g. each try/catch separately

Reply:
See updated methods below.